test/index.js

```diff
@@ -0,0 +1,159 @@
+  const clientConfig = {
+    appId: 'HyhaWO1SG',
+    prvKey: HEX_PRVKEY_NIST,
+    appSecret: SECRET,
+    api: API,
+    env: STAGE,
+  };
+  const partialConfig = {
+    appId: 'anId',
+    appSecret: 'anAppSecret',
+    prvKey: 'aPrivateKey'
+  };
+
+  describe('newClient', function newClientTest() {
+    this.beforeEach(() => {
+      sinon.stub(jwtjs, 'verify').returns(true);
+  
+      mockAuthCode(authCode, returnData, true);
+    });
+  
+    this.afterEach(() => {
+      nock.cleanAll();
+      jwtjs.verify.restore();
+    });
+
+    it('expects an application id', () => {
+      try {
+        sipClient.newClient();
+      } catch (error) {
+        expect(error.message).to.equal('Please supply your application ID.');
+      }
+    });
+
+    it('expects an application secret', () => {
+      try {
+        sipClient.newClient({ appId: 'something' });
+      } catch (error) {
+        expect(error.message).to.equal('Please supply your application secret.');
+      }
+    });
+
+    it('expects an application private key', () => {
+      try {
+        sipClient.newClient({ appId: 'something', appSecret: 'a secret' });
+      } catch (error) {
+        expect(error.message).to.equal('Please supply your application private key.');
+      }
+    });
+
+    it('should allow a minimum config', () => {
+      const client = sipClient.newClient(partialConfig);
+      expect(client.exchangeCode).to.be.a('function');
+    });
+
+    it('should allow a config with optional keys', () => {
+      const client = sipClient.newClient(clientConfig);
+      expect(client.exchangeCode).to.be.a('function');
+    });
+
+    it('should exchange an encrypted code', (done) => {
+      const doneFn = done;
+      const client = sipClient.newClient(clientConfig);
+      client.exchangeCode(authCode).then((data) => {
+        // console.log(data);
+        assert.equal(data.data[1].label, 'contact.personal.phoneNumber', 'The labels are not equal');
+        assert.isTrue(data.data[1].isOwner, 'isOwner not true');
+        assert.isTrue(data.data[1].isValid, 'isValid not true');
+        doneFn();
+      })
+        .catch((error) => {
+          doneFn(error);
+        });
+    });
+
+    it('should not exchange data that is not encrypted', (done) => {
+      const doneFn = done;
+      const client = sipClient.newClient(clientConfig);
+
+      nock.cleanAll();
+      mockAuthCode(authCode, returnData, false);
+
+      client.exchangeCode(authCode).then((data) => {
+        expect(data.data).to.equal(undefined);
+        doneFn();
+      })
+        .catch((error) => {
+          doneFn(error);
+        });
+    });
+
+    it('should handle errors from the authCode end point', (done) => {
+      const doneFn = done;
+      const client = sipClient.newClient(clientConfig);
+      const errorMessage = 'There was an error';
+
+      nock.cleanAll();
+      mockAuthCodeThrowErrror(authCode, errorMessage);
```

Comment:
```mockAuthCodeThrowError``` requires 1 argument but 2 given. Perhaps pass in errorMessage to your function. Not a breaking change so not a biggy

Reply:
Also spelling error :) ```mockAuthCodeThrowErrror```

Reply:
```mockAuthCodeThrowError```, there is a triple 'r'